src/workloads/docs/SamplingLoader.yml

```diff
@@ -0,0 +1,112 @@
+SchemaVersion: 2018-07-01
+Owner: "@mongodb/query"
+Description: |
+  This workload demonstrates using the SamplingLoader to expand the contents of a collection. This
+  actor will take a random sample of a configurable size and re-insert the same documents. It will
+  project out the "_id" field before re-inserting to avoid duplicate key errors.
+GlobalDefaults:
+  db: &db test
+  Collection: &Collection sampling_loader_demo
+
+Actors:
+# This is just to seed the collection with 40 documents so that we'll have some sample data to draw
+# from.
+- Name: CollectionSeed
+  Type: MonotonicSingleLoader
+  Phases:
+  - Collection: *Collection
+    Database: *db
+    Repeat: 1
+    Document:
+      x: {^Inc: {}}
+    DocumentCount: 40
+    BatchSize: 10
+  - {Nop: true}
+  - {Nop: true}
+  - {Nop: true}
+
+# Our first demonstration of the sampling loader.
+- Name: ImpliedSampleSizeDemo
+  Type: SamplingLoader
+  Threads: 1
+  Phases:
+  - {Nop: true}
+  - Repeat: 1
+    Database: *db
+    Collection: *Collection
+    # Number of documents in a single insert operation.
+    InsertBatchSize: 5
+    # Number of insert operations. The total number of documents to be inserted is
+    # Batches * InsertBatchSize.
+    Batches: 2
+    # Number of documents in the sample. The loader will cycle through the sampled docs, using them
+    # as templates # for the ones to be incerted, until it inserts the requested number of batches.
+    # For example, if the sample size is set to 1, exactly one document is sampled from the original
+    # dataset and reused as a template for  _all_ of the inserted docs. And if the sample size is
+    # equal to the insert batch size, each sampled document will be "re-inserted" as many times as
+    # the number of batches.
+    #
+    # This may be useful if you want to insert a lot of documents to measure that performance, but
```

Comment:
Thinking more about this comment: are we at all concerned that the perf of sampling will be baked into the measurements for the actor's insert action? As you say, for large number of inserted documents the cost of sampling will amortize, but for "point" inserts this might be an issue?

Actually, I'm not clear how Genny measures a complex action like this one. There will be multiple commands executed against the server in a single phase of the actor -- what is latency of the phase in this case?

Reply:
Done the re-word!

We don't need to worry about the sample counting towards the metrics - the metrics are calculated via these calls to 'start()': https://github.com/mongodb/genny/pull/766/files/885799174853bb0497202e47a454c4f2b4794d9a#diff-57d23b69c3558547e64c4613f5164398d94531875d280bf32d376fe07bea65c3R124

Reply:
Oh, cool! I didn't realize this is how the metrics are controlled by the actor.

Reply:
Actually, don't see the reword in 192d25e yet

Reply:
Oops! Forgot to hit save before committing 🤦🏼